src/templates/field.type.ts

```diff
@@ -1,8 +1,15 @@
 import { Field } from './field';
-import { OnInit, OnChanges, DoCheck, AfterContentInit, AfterContentChecked, AfterViewInit, OnDestroy, AfterViewChecked } from '@angular/core';
+import { OnInit, OnChanges, DoCheck, AfterContentInit, AfterContentChecked, AfterViewInit, OnDestroy, AfterViewChecked, Renderer, ElementRef } from '@angular/core';
 
 export abstract class FieldType extends Field implements OnInit, OnChanges, DoCheck, AfterContentInit, AfterContentChecked, AfterViewInit, AfterViewChecked, OnDestroy {
-
+  private _renderer: Renderer;
+  private _elementRef: ElementRef;
+  private attrbites = ['placeholder', 'tabindex', 'step'];
+  constructor(renderer: Renderer, elementRef: ElementRef) {
```

Comment:
I don't prefer this way :), instead use a custom directive like we have been doing for focus `formlyNgFocus`